code/modules/antagonists/demon/demons.dm

```diff
@@ -0,0 +1,172 @@
+#define SIN_GLUTTONY "gluttony"
+#define SIN_GREED "greed"
+#define SIN_SLOTH "sloth"
+#define SIN_WRATH "wrath"
+#define SIN_ENVY "envy"
+#define SIN_PRIDE "pride"
+
+/mob/living/carbon/human/Life()
+	. = ..()
+	if(is_sinfuldemon(src))
+		var/datum/antagonist/sinfuldemon/demon = mind.has_antag_datum(/datum/antagonist/sinfuldemon)
+		demon.sinfuldemon_life()
+
+/datum/antagonist/sinfuldemon
+	name = "Sinful Demon"
+	roundend_category = "demons of sin"
+	antagpanel_category = "Demon"
+	job_rank = ROLE_SINFULDEMON
+	show_to_ghosts = TRUE
+	var/demonsin
+	var/static/list/demonsins = list(SIN_GLUTTONY,SIN_GREED,SIN_WRATH,SIN_ENVY,SIN_PRIDE)
+	var/static/list/demon_spells = typecacheof(list(
+		/obj/effect/proc_holder/spell/targeted/shapeshift/demon,
+		/obj/effect/proc_holder/spell/targeted/shapeshift/demon/gluttony,
+		/obj/effect/proc_holder/spell/targeted/shapeshift/demon/wrath,
+		/obj/effect/proc_holder/spell/targeted/forcewall/gluttony,
+		/obj/effect/proc_holder/spell/aoe_turf/conjure/summon_greedslots,
+		/obj/effect/proc_holder/spell/targeted/inflict_handler/ignite,
+		/obj/effect/proc_holder/spell/targeted/touch/envy,
+		/obj/effect/proc_holder/spell/aoe_turf/conjure/summon_mirror))
+
+/datum/antagonist/sinfuldemon/proc/sinfuldemon_life()
+	var/mob/living/carbon/C = owner.current
+	if(!C)
+		return
+	if(istype(get_area(C.loc), /area/chapel))
+		demon_burn()
+
+/datum/antagonist/sinfuldemon/proc/demon_burn() //sinful demons are even more vulnerable to the chapel than vampires, but can turn into their true form to negate this.
+	var/mob/living/L = owner.current
+	if(!L)
+		return
+	if(L.stat != DEAD) //demons however wont dust from the chapel so this needs to be a check to avoid spam while they're already dead
+		if(prob(50) && L.health >= 50)
+			switch(L.health)
+				if(85 to 100)
+					L.visible_message(span_warning("[L]'s skin begins to heat up and darken!"), span_danger("Your flesh begins to sear..."))
+				if(60 to 85)
+					L.visible_message(span_warning("[L]'s skin begins to melt apart!"), span_danger("Your skin is melting!"), "You hear sizzling.")
+			L.adjustFireLoss(5)
+		else if(L.health < 60)
+			if(!L.on_fire)
+				L.visible_message(span_warning("[L] lights up in a holy blaze!"), span_danger("Your skin catches fire!"))
+				L.emote("scream")
+			else
+				L.visible_message(span_warning("[L] continues to burn!"), span_danger("You continue to burn!"))
```

Comment:
won't this just keep spamming while they're alive and in the chapel?

Reply:
yes, it is also what happens to vamps. It can fuck you up pretty fast so I thought to keep it.